ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -145,7 +146,7 @@ public int appendFrom(ByteBuffer buffer) throws IOException {
               + capacityInBytes + "]");
     } else {
       while (buffer.hasRemaining()) {
-        bytesWritten += fileChannel.write(buffer, endOffset.get());
+        bytesWritten += fileChannel.write(buffer, endOffset.get() + bytesWritten);
```

Comment:
Reason for this change?

Reply:
Looks like a bug, right? 

`endOffset` doesn't change and If `fileChannel.wirte()` can't finish in one iteration, it always write at the position `endOffset.get()`.

Reply:
Just curious, if it is a bug, we should have a test case for it,right?

Reply:
You are right, there should be a test and mock the fileChannel, however this is not straight forward and what we really interested in is fileChannel itself. 

This change has been pushed in #1000 because it may impact compaction.